cabal-install/src/Distribution/Client/Init/Interactive/Command.hs

```diff
@@ -467,3 +467,8 @@ srcDirsPrompt flags = getSrcDirs flags $ do
       True
 
     return [dir]
+
+promptOrDefault :: Interactive m => Maybe String -> String -> m String
+promptOrDefault m s = case m of
```

Comment:
Instead of  `case` you could use `maybe (promptStr s MandatoryPrompt)  (promptStr s . DefaultPrompt) m`. And can remove (eta-reduce) `m` if the order of the arguments is flipped, as suggested above.